daemon/algod/api/server/v2/utils.go

```diff
@@ -110,18 +110,12 @@ func omitEmpty[T comparable](val T) *T {
 	return &val
 }
 
-func byteOrNil(data []byte) *[]byte {
-	if len(data) == 0 {
-		return nil
-	}
-	return &data
-}
-
-func nilToZero(numPtr *uint64) uint64 {
-	if numPtr == nil {
-		return 0
+func nilToDefault[T any](valPtr *T) T {
```

Comment:
I like the generic version, but I don't know if you need to change the name from `nilToZero`. In Go is common to talk about the "zero value" of a struct, rather than a "default value".

Reply:
right, changed to `nilToZero`.